packages/aws-cdk/lib/api/hotswap/common.ts

```diff
@@ -33,6 +39,43 @@ export enum ChangeHotswapImpact {
 
 export type ChangeHotswapResult = HotswapOperation | ChangeHotswapImpact;
 
-export function assetMetadataChanged(change: cfn_diff.ResourceDifference): boolean {
+/**
+ * Represents a change that can be hotswapped.
+ */
+export class HotswappableResourceChange {
+  /**
+   * The value the resource is being updated to.
+   */
+  newValue: cfn_diff.Resource;
+
+  /**
+   * The changes made to the resource properties.
+   */
+  propertyUpdates: { [key: string]: cfn_diff.PropertyDifference<any> };
+
+  public constructor(newValue: cfn_diff.Resource, propertyUpdates: { [key: string]: cfn_diff.PropertyDifference<any> }) {
+    this.newValue = newValue;
+    this.propertyUpdates = propertyUpdates;
+  }
+}
+
+export async function establishHotswappableResourceName(
+  logicalId: string, evaluateCfnTemplate: EvaluateCloudFormationTemplate, nameInCfnTemplate: any,
```

Comment:
Order is a little weird here.... I would put `EvaluateCloudFormationTemplate` as the last parameter.